pkg/compactor/compactor.go

```diff
@@ -150,22 +148,22 @@ func (cp *Compactor) Compact(ctx context.Context, opts *brtypes.CompactOptions)
 		}
 	}
 
-	// Then take snapeshot of ETCD
+	// Then take snapshot of ETCD
 	snapshotReqCtx, cancel := context.WithTimeout(ctx, opts.SnapshotTimeout.Duration)
 	defer cancel()
 
 	// Determine suffix of compacted snapshot that will be result of this compaction
-	suffix := cmpctOptions.BaseSnapshot.CompressionSuffix
-	if len(cmpctOptions.DeltaSnapList) > 0 {
-		suffix = cmpctOptions.DeltaSnapList[cmpctOptions.DeltaSnapList.Len()-1].CompressionSuffix
+	suffix := compactorRestoreOptions.BaseSnapshot.CompressionSuffix
+	if len(compactorRestoreOptions.DeltaSnapList) > 0 {
+		suffix = compactorRestoreOptions.DeltaSnapList[compactorRestoreOptions.DeltaSnapList.Len()-1].CompressionSuffix
 	}
 
 	isCompressed, compressionPolicy, err := compressor.IsSnapshotCompressed(suffix)
```

Comment:
I don't think this is essential. When etcd-backup-restore is the one that's compressing the snapshots and ensuring that it adds the correct compression suffix to the snapshot name, then why worry about checking the file contents to figure out what compression algorithm was used to compress the snapshot, when etcd-backup-restore can simply check the compression suffix that it itself had added during snapshotting?

Also, the same logic would then apply to parsing snapshots, ie, etcdbr would have to start checking the contents of the snapshot to figure out what event revisions exist in the snapshot, when it can simply get this information from the snapshot name like it does today.

Again, all I've done in compactor package is change the variable names, nothing else.